scripts/bumpfee.py

```diff
@@ -0,0 +1,264 @@
+from jmclient.cli_options import OptionParser, add_base_options
+import jmbitcoin as btc
+import sys
+
+jlog = get_log()
+parser = OptionParser(
+    usage='usage: %prog [options] [wallet file] txid',
+    description=
+    'Bumps the fee on a wallet transaction')
+parser.add_option('--psbt',
+                  action='store_true',
+                  dest='with_psbt',
+                  default=False,
+                  help='output as psbt instead of '
+                  'broadcasting the transaction.')
+parser.add_option('-f',
+                  '--txfee',
+                  action='store',
+                  type='int',
+                  dest='txfee',
+                  default=-1,
+                  help='Bitcoin miner tx_fee to use for transaction(s). A number higher '
+                  'than 1000 is used as "satoshi per KB" tx fee. A number lower than that '
+                  'uses the dynamic fee estimation of your blockchain provider as '
+                  'confirmation target. This temporarily overrides the "tx_fees" setting '
+                  'in your joinmarket.cfg. Works the same way as described in it. Check '
+                  'it for examples.')
+parser.add_option('-a',
+                  '--amtmixdepths',
+                  action='store',
+                  type='int',
+                  dest='amtmixdepths',
+                  help='number of mixdepths in wallet, default 5',
+                  default=5)
+parser.add_option('-g',
+                  '--gap-limit',
+                  type="int",
+                  action='store',
+                  dest='gaplimit',
+                  help='gap limit for wallet, default=6',
+                  default=6)
+parser.add_option('--yes',
+                  action='store_true',
+                  dest='answeryes',
+                  default=False,
+                  help='answer yes to everything')
+add_base_options(parser)
+
+def check_valid_candidate(orig_tx, wallet):
+    orig_tx_info = jm_single().bc_interface.get_transaction(orig_tx.GetTxid()[::-1])
+    # check that the transaction is still unconfirmed
+    if orig_tx_info['confirmations'] > 0:
+        raise RuntimeWarning('Transaction already confirmed. Nothing to do.')
+
+    # all transaction inputs must belong to the wallet
+    own_inputs_n = len(wallet.inputs_consumed_by_tx(orig_tx))
+    tx_inputs_n = len(orig_tx.vin)
+
+    if own_inputs_n != tx_inputs_n:
+        raise ValueError('Transaction inputs should belong to the wallet.')
+
+    # at least one input should signal opt-in rbf
+    if not any([vin.nSequence <= 0xffffffff - 2 for vin in orig_tx.vin]):
+        raise ValueError('Transaction not replaceable.')
+
+    # wallet should own at least one output that we can change values for
+    if not any(
+        [wallet.is_known_script(vout.scriptPubKey) for vout in orig_tx.vout]
+    ):
+        raise ValueError('Transaction has no output we can deduct fees from.')
```

Comment:
Are you sure this restriction is needed?

What if, for example, I am sending coins from this wallet to another wallet I own, and maybe (it's pretty common) in a 'sweep' type transaction with no change?

Reply:
No. This restriction is not needed; however, the issue becomes deciding which output to deduct the additional fees from. An exception may be made for a sweep transaction where only one output exists. Are there any other use-cases where deducting the fees from a non-wallet output would make sense?

Reply:
Well if the destination of the payment is owned by the sender, then they may (for whatever reason) want to deduct the extra fee from one of the destination outputs. It's much less likely (compared to the sweep case), but there doesn't seem to be a good reason to prevent it?

It feels like the script should be using an algorithm of "if not specified, use the 1 change output as the sane default, but if specified, or if there is not 1 change output, require the user to choose the output".

Reply:
The script now implements this logic.